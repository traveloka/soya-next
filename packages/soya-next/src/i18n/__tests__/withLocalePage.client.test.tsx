import React from "react";
import { render } from "@testing-library/react";
import setupTest from "./utils/withLocalePageSetup";

describe("withLocalePage - Browser", () => {
  it("should add default locale, locale, and site locales to page props", async () => {
    const { commonProps, context, PageWithLocale } = setupTest();
    window.__NEXT_DATA__ = {
      props: context,
    } as any;
    const props = await PageWithLocale.getInitialProps(commonProps);
    const { getByTestId } = render(<PageWithLocale {...props} />);
    const componentLocalePropsDiv = getByTestId("component-locale-props");
    expect(componentLocalePropsDiv.textContent).toMatchSnapshot();
  });

  it("should update locale if supported", async () => {
    const { commonProps, context, PageWithLocale } = setupTest();
    window.__NEXT_DATA__ = {
      props: context,
    } as any;
    const props = await PageWithLocale.getInitialProps({
      ...commonProps,
      query: {
        locale: "en-id",
      },
    });
    const { getByTestId } = render(<PageWithLocale {...props} />);
    const componentLocalePropsDiv = getByTestId("component-locale-props");
    expect(componentLocalePropsDiv.textContent).toMatchSnapshot();
  });

  it("should not update locale if unsupported", async () => {
    const { commonProps, context, PageWithLocale } = setupTest();
    window.__NEXT_DATA__ = {
      props: context,
    } as any;
    const props = await PageWithLocale.getInitialProps({
      ...commonProps,
      query: {
        locale: "en-sg",
      },
    });
    const { getByTestId } = render(<PageWithLocale {...props} />);
    const componentLocalePropsDiv = getByTestId("component-locale-props");
    expect(componentLocalePropsDiv.textContent).toMatchSnapshot();
  });
});
