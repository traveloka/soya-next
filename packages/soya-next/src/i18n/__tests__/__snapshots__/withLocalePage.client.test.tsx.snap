// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLocalePage - Browser should add default locale, locale, and site locales to page props 1`] = `
"{
  \\"init\\": true,
  \\"defaultLocale\\": \\"id-id\\",
  \\"siteLocales\\": [
    \\"id-id\\",
    \\"en-id\\"
  ],
  \\"locale\\": {
    \\"language\\": \\"id\\",
    \\"country\\": \\"id\\"
  }
}"
`;

exports[`withLocalePage - Browser should not update locale if unsupported 1`] = `
"{
  \\"init\\": true,
  \\"defaultLocale\\": \\"id-id\\",
  \\"siteLocales\\": [
    \\"id-id\\",
    \\"en-id\\"
  ],
  \\"locale\\": {
    \\"language\\": \\"id\\",
    \\"country\\": \\"id\\"
  }
}"
`;

exports[`withLocalePage - Browser should update locale if supported 1`] = `
"{
  \\"init\\": true,
  \\"defaultLocale\\": \\"id-id\\",
  \\"siteLocales\\": [
    \\"id-id\\",
    \\"en-id\\"
  ],
  \\"locale\\": {
    \\"language\\": \\"en\\",
    \\"country\\": \\"id\\"
  }
}"
`;
